app/src/main/java/com/hadiyarajesh/notex/reminder/MainNotificationWorker.kt

```diff
@@ -0,0 +1,31 @@
+package com.hadiyarajesh.notex.reminder
+
+import android.content.Context
+import com.hadiyarajesh.notex.database.entity.Reminder
+import java.time.Instant
+
+interface MainNotificationWorker {
+
+
+    fun createWorkRequestAndEnqueue(
+        context: Context,
+        reminder: Reminder?,
+        requestType: RequestType = RequestType.OneTimeRequest,
+        isFirstTime: Boolean = true
+    )
+
+    fun postponeRequestAndEnqueue(
+        context: Context,
+        reminderId: Long,
+        time: Instant
+    )
+
+    fun cancelWorkRequest(context: Context, tag: String): Boolean
+
+    companion object {
+        enum class RequestType {
+            OneTimeRequest,
+            PeriodicRequest
+        }
```

Comment:
We can use a common package for defining these ENUMS.

It's not better to Define it here. It will restrict the usage of this ENUMS with the Class Object only.